src/base/bittorrent/torrenthandle.cpp

```diff
@@ -74,16 +74,27 @@ const QString QB_EXT {QStringLiteral(".!qB")};
 
 using namespace BitTorrent;
 
+#if (LIBTORRENT_VERSION_NUM >= 10200)
+namespace libtorrent
+{
+    namespace aux
+    {
+        inline uint qHash(const strong_typedef<std::int32_t, file_index_tag> &key, uint seed)
+        {
+            return ::qHash(underlying_index_t<strong_typedef<std::int32_t, file_index_tag>>::type {key}, seed);
```

Comment:
will this work? `::qHash(underlying_index_t<lt::file_index_t>::type {key}, seed);`

Reply:
or perhaps (didn't test):
`return static_cast<uint>(std::hash<lt::file_index_t>{}() ^ seed);`

Reply:
>or perhaps (didn't test):
`return static_cast<uint>(std::hash<lt::file_index_t>{}() ^ seed);`

Does libtorrent provide `std::hash` specializations for its custom index types?
Or do you want me to implement it as well? It is certainly superfluous!

Reply:
>Does libtorrent provide std::hash specializations for its custom index types?

Yes, IMO it would be best if we use it.

Reply:
>Yes, IMO it would be best if we use it.

Ok.

Reply:
PR updated.

Reply:
maybe you would like to make it handle all types derived from `strong_typedef` as such:
```c++
namespace libtorrent
{
    namespace aux
    {
        template <typename ...Ts>
        uint qHash(const strong_typedef<Ts...> &key, const uint seed)
        {
            return static_cast<uint>((std::hash<strong_typedef<Ts...>> {})(key)) ^ seed;
        }
    }
}
```
this way we don't need to bother with `<std::int32_t, file_index_tag>` specialization.

Reply:
Ok, changed.